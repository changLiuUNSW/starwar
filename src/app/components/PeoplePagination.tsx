import { PaginationItem, Pagination, PaginationLink, Spinner } from 'reactstrap';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
`;

const SpinnerWrapper = styled.div`
  margin-top: 0.2rem;
  margin-left: 1rem;
`;

interface Props {
  loading?: boolean;
  onNext?: () => void;
  onPrev?: () => void;
}

const PeoplePagination = ({ onNext, onPrev, loading }: Props) => (
  <Container>
    <Pagination aria-label="people table pagination">
      <PaginationItem disabled={loading || !onPrev}>
        <PaginationLink href="#" onClick={onPrev}>
          Previous
        </PaginationLink>
      </PaginationItem>
      <PaginationItem disabled={loading || !onNext}>
        <PaginationLink href="#" onClick={onNext}>
          Next
        </PaginationLink>
      </PaginationItem>
    </Pagination>
    {loading && (
      <SpinnerWrapper>
        <Spinner color="primary" size="md" />
      </SpinnerWrapper>
    )}
  </Container>
);

export default PeoplePagination;
