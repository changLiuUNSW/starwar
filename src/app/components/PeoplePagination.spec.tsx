import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import PeoplePagination from './PeoplePagination';

it('PeoplePagination renders correctly', () => {
  const tree = renderer.create(<PeoplePagination />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('PeoplePagination should show spinner and disable both buttons if loading', () => {
  const onNext = jest.fn();
  const onPrev = jest.fn();
  render(<PeoplePagination loading={true} onNext={onNext} onPrev={onPrev} />);
  expect(screen.getAllByRole('listitem')[0]).toHaveClass('disabled');
  expect(screen.getAllByRole('listitem')[1]).toHaveClass('disabled');
  expect(screen.getByRole('status')).not.toBeNull();
});

it('PeoplePagination both button should be called if not loading', () => {
  const onNext = jest.fn();
  const onPrev = jest.fn();
  render(<PeoplePagination loading={false} onNext={onNext} onPrev={onPrev} />);
  fireEvent.click(screen.getByText('Previous'));
  fireEvent.click(screen.getByText('Next'));
  expect(onPrev).toHaveBeenCalled();
  expect(onNext).toHaveBeenCalled();
});
