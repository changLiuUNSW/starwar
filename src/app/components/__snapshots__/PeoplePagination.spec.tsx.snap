// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeoplePagination renders correctly 1`] = `
<div
  className="sc-bdfBwQ gbvaeM"
>
  <nav
    aria-label="people table pagination"
    className=""
  >
    <ul
      className="pagination"
    >
      <li
        className="page-item disabled"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          Previous
        </a>
      </li>
      <li
        className="page-item disabled"
      >
        <a
          className="page-link"
          href="#"
          onClick={[Function]}
        >
          Next
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
